### PR TITLE
Update v4-api.md

### PR DESCRIPTION
**Pull request by marikaner**
Add new insights about expansion. It probably makes sense for everyone in the team to take note of that. @FrankEssenberger @mr-flannery @jjtang1985

### CODE REVIEW

File: knowledge-base/v4-api.md [hunk 1]
```diff
@@ -19,10 +19,11 @@ This document contains a collection of OData v4 features we want to implement an
...
 
 ## $expand with Subqueries
-- Make queries on expanded entities (navigation properties)
+- Make queries on expanded entities (navigation properties and complex properties)
 - Works for `getAll` and `getByKey`
-- Example request for collection: https://services.odata.org/TripPinRESTierService/(S(ljgcbxqwp45c5l5m0h24kk1g))/People?$select=Friends&$expand=Friends($select=UserName,Emails;$filter=startswith(UserName,%27s%27))
-- Example request for single link: https://services.odata.org/TripPinRESTierService/(S(ljgcbxqwp45c5l5m0h24kk1g))/People?$select=BestFriend&$expand=BestFriend($select=UserName,Emails)
+- Example request for one to many link: https://services.odata.org/TripPinRESTierService/(S(ljgcbxqwp45c5l5m0h24kk1g))/People?$select=Friends&$expand=Friends($select=UserName,Emails;$filter=startswith(UserName,%27s%27))
+- Example request for one to one link: https://services.odata.org/TripPinRESTierService/(S(ljgcbxqwp45c5l5m0h24kk1g))/People?$select=BestFriend&$expand=BestFriend($select=UserName,Emails)
+- Could not find a working example for complex types
```
Comment (mr-flannery) [inline knowledge-base/v4-api.md:26]: Where did you pick up from that it's a possibility? the spec?

File: knowledge-base/v4-api.md [hunk 2]
```diff
@@ -79,8 +80,7 @@ This document contains a collection of OData v4 features we want to implement an
   ```
 
 ### Decision:
-- Second proposal accepted under the assumption that expanding navigation properties automatically selects all properties of the navigation property.
-- If the assumtion is incorrect, use the first proposal.
+- First proposal accepted. Our assumption that expanding navigation properties automatically selects all properties of the navigation property has proven correct.
```
Comment (mr-flannery) [inline knowledge-base/v4-api.md:83]: If our assumption is correct, shouldn't the second one then be the accepted one, not the first?

Comment (FrankEssenberger) [reply knowledge-base/v4-api.md:83]: 👍 also confused. Question which just came to my mind. What happens for nested navigation properties so if the expanded one has again a navigation property? I guess this is not expanded, right?

File: knowledge-base/v4-api.md (hunk 1 above)
Comment (marikaner) [reply knowledge-base/v4-api.md:26]: I talked to the Java colleagues and now also crosschecked that in the spec: [Example 115](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#sec_SystemQueryOptionexpand)

File: knowledge-base/v4-api.md (hunk 2 above)
Comment (marikaner) [reply knowledge-base/v4-api.md:83]: It can also be expanded. And of course you are right, I confused the order.

**Comment by marikaner** [comment]
@FrankEssenberger @jjtang1985

**Pull request merged**